pkg/quarkus/v1alpha/api.go

```diff
@@ -116,3 +149,28 @@ func (p *createAPISubcommand) InjectResource(res *resource.Resource) error {
 
 	return nil
 }
+
+const (
+	makefileBundleVarFragment = `
```

Comment:
Just curious, if a user were to run `create api` multiple times would the make file only have the latest resources created as a part of `create api` set to be included in the bundle?

For example, if I ran in this sequence what would the Makefile look like?:
1. `operator-sdk init --plugins java --domain example.com`
2. `operator-sdk create api --plugins java --group crew --version v1alpha1 --kind Captain`
3. `operator-sdk create api --plugins java --group crew --version v1alpha1 --kind Sailor`

I would expect that it should look something like:
```make
PHONY: bundle
bundle:  ## Generate bundle manifests and metadata, then validate generated files.
	cat target/kubernetes/captains.crew.example.com-v1alpha1.yml target/kubernetes/kubernetes.yml  target/kubernetes/sailors.crew.example.com-v1alpha1.yml target/kubernetes/kubernetes.yml | operator-sdk generate bundle -q --overwrite --version 0.1.1 --default-channel=stable --channels=stable --package=projectName
	operator-sdk bundle validate ./bundle
```

Reply:
Expected here is:

```
PHONY: bundle
bundle:  ## Generate bundle manifests and metadata, then validate generated files.
	cat target/kubernetes/sailors.crew.example.com-v1alpha1.yml target/kubernetes/kubernetes.yml | operator-sdk generate bundle -q --overwrite --version 0.1.1 --default-channel=stable --channels=stable --package=projectName
	operator-sdk bundle validate ./bundle
```

Reply:
The current logic does not handle multiple create API commands. @jmrodri - is this functionality necessary, would like to know your thoughts.

Reply:
@everettraven - I just tested this scenario and found that this case is not handled for other scaffolding files as well.
Output is shown below:

```
.
├── Makefile
├── PROJECT
├── pom.xml
├── src
│   └── main
│       ├── java
│       │   └── com
│       │       └── example
│       │           ├── Captain.java
│       │           ├── CaptainReconciler.java
│       │           ├── CaptainSpec.java
│       │           ├── CaptainStatus.java
│       │           ├── Sailor.java
│       │           ├── SailorReconciler.java
│       │           ├── SailorSpec.java
│       │           └── SailorStatus.java
│       └── resources
│           └── application.properties
└── test

7 directories, 12 files
```

You can see multiple files got generated for two different `kind`.

Reply:
Also, use the below command as java is not enabled yet. We are scaffolding the `quarkus` plugin.

```
operator-sdk init --plugins quarkus --domain example.com
operator-sdk create api --plugins quarkus --group crew --version v1alpha1 --kind Captain
operator-sdk create api --plugins quarkus --group crew --version v1alpha1 --kind Sailor
```

Reply:
> I just tested this scenario and found that this case is not handled for other scaffolding files as well. Output is shown below:
> 
> ```
> .
> ├── Makefile
> ├── PROJECT
> ├── pom.xml
> ├── src
> │   └── main
> │       ├── java
> │       │   └── com
> │       │       └── example
> │       │           ├── Captain.java
> │       │           ├── CaptainReconciler.java
> │       │           ├── CaptainSpec.java
> │       │           ├── CaptainStatus.java
> │       │           ├── Sailor.java
> │       │           ├── SailorReconciler.java
> │       │           ├── SailorSpec.java
> │       │           └── SailorStatus.java
> │       └── resources
> │           └── application.properties
> └── test
> 
> 7 directories, 12 files
> ```
> 
> You can see multiple files got generated for two different `kind`.

This output is exactly what I would expect when running multiple `create api` subcommands.


> Expected is here:
> 
> ```
> PHONY: bundle
> bundle:  ## Generate bundle manifests and metadata, then validate generated files.
> 	cat target/kubernetes/sailors.crew.example.com-v1alpha1.yml target/kubernetes/kubernetes.yml | operator-sdk generate bundle -q --overwrite --version 0.1.1 --default-channel=stable --channels=stable --package=projectName
> 	operator-sdk bundle validate ./bundle
> ```

This seems to be different from what I would expect after running `create api` more than once. I would think that a new file would be added to the `cat` command based on the GVK that is used in each subsequent `create api` run. 

So I would think we should make it so that this would be the command that is run:
```make
PHONY: bundle
bundle:  ## Generate bundle manifests and metadata, then validate generated files.
	cat target/kubernetes/captains.crew.example.com-v1alpha1.yml target/kubernetes/sailors.crew.example.com-v1alpha1.yml target/kubernetes/kubernetes.yml | operator-sdk generate bundle -q --overwrite --version 0.1.1 --default-channel=stable --channels=stable --package=projectName
	operator-sdk bundle validate ./bundle
```

To potentially make it easier to read the make target and understand what is happening (and potentially making it easier to append a new file to it) we might want to do something like:
```make
PHONY: bundle
bundle: ## Generate bundle manifests and metadata, then validate generated files.
      cat target/kubernetes/captains.crew.example.com-v1alpha1.yml \
      target/kubernetes/sailors.crew.example.com-v1alpha1.yml \
      target/kubernetes/kubernetes.yml \
      | operator-sdk generate bundle -q --overwrite --version 0.1.1 --default-channel=stable --channels=stable --package=projectName
	operator-sdk bundle validate ./bundle
```

Also after running the commands and building the project this is what the tree of `target/kubernetes` looks like:
```
target/kubernetes
├── captains.crew.example.com-v1.yml
├── kubernetes.json
├── kubernetes.yml
└── sailors.crew.example.com-v1.yml
```
One thing I noticed is even after explicitly setting the version to be `v1alpha1` it still seems to put the version on these targets as `v1`. Is this a bug?

@laxmikantbpandhare what do you think?

Reply:
Not really, I can see the `v1` version when I generated artifacts.

```
.
├── kubernetes.json
├── kubernetes.yml
└── memcacheds.cache.example.com-v1.yml

0 directories, 3 files
```

Reply:
> Not really, I can see the `v1` version when I generated artifacts.
> 
> ```
> .
> ├── kubernetes.json
> ├── kubernetes.yml
> └── memcacheds.cache.example.com-v1.yml
> 
> 0 directories, 3 files
> ```
> 
> and Makefile has below details
> 
> ```
> ##@Bundle
> .PHONY: bundle
> bundle:  ## Generate bundle manifests and metadata, then validate generated files.
> 	cat target/kubernetes/memcacheds.cache.example.com-v1.yml target/kubernetes/kubernetes.yml | operator-sdk generate bundle -q --overwrite --version 0.1.1 --default-channel=stable --channels=stable --package=memcached-quarkus-operator
> 	operator-sdk bundle validate ./bundle
> ```

But does it still use v1 in `target/kubernetes` if you specify `--version v1alpha1` when running `create api`?

> I don't think we are going to deal with multiple `create api` commands and never come across such a scenario.

Why wouldn't we come across this scenario? If a user wants to create multiple different Kinds in the same Group and Version will they be required to create a new Quarkus operator project for each CRD?

This same scenario seems possible in the Go and Ansible plugin and is actually done in some testdata sample generation logic.

For the Go plugin we can see in Kubebuilder this is covered by generating a sample here:
https://github.com/kubernetes-sigs/kubebuilder/blob/a2b2be2afbf8b1e3068dd09927f01944b7b31f54/test/testdata/generate.sh#L107-L109

For the Ansible plugin we can see it is covered in this sample here:
https://github.com/operator-framework/operator-sdk/blob/fe30cc21d09174c76edf89f04416b753c7d66701/hack/generate/samples/internal/ansible/advanced_molecule.go#L555-L588

Reply:
Let's wait for @jmrodri comment. We will discuss this further.

Reply:
@everettraven - thanks for bringing this. I had a discussion with Jesus and we are trying to fix this. I will update here soon.

Reply:
Modified the code. Please check once and let me know your thoughts.